### PR TITLE
Feature/complete flow steps

### PR DESCRIPTION
**Pull request by mikeSmith08**
- Fix application context so that Spring and Cucumber play nicely with multiple step definition files.
 - Added base components for input fields, radio buttons and submit buttons.
 - Continued complete flow up to and including the enter DOB page.

### CODE REVIEW

File: src/test/java/uk/gov/dhsc/htbhf/page/DoYouHaveChildren.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package uk.gov.dhsc.htbhf.page;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import uk.gov.dhsc.htbhf.page.component.RadioButton;
+import uk.gov.dhsc.htbhf.page.component.SubmitButton;
+
+/**
+ * Page object for the Do You Live In Scotland page.
```
Comment (YetAnotherMatt) [inline src/test/java/uk/gov/dhsc/htbhf/page/DoYouHaveChildren.java]: Javadoc doesn't match class name

Comment (mikeSmith08) [reply src/test/java/uk/gov/dhsc/htbhf/page/DoYouHaveChildren.java]: Fixed.

File: src/test/java/uk/gov/dhsc/htbhf/page/BasePage.java [hunk 2]
```diff
@@ -33,16 +37,16 @@ public void open() {
...
-    public WebElement findByCss(String cssSelector) {
-        return webDriver.findElement(By.cssSelector(cssSelector));
+    public List<WebElement> getRadioButtons() {
+        return webDriver.findElements(By.className("govuk-radios__item"));
     }
 
-    public void waitForPageToLoad() {
-        wait.until(ExpectedConditions.titleIs(getPageTitle()));
+    public List<WebElement> getAllRadioLabels(String option) {
+        return webDriver.findElements(By.className("govuk-label govuk-radios__label"));
```
Comment (gethyn1) [inline src/test/java/uk/gov/dhsc/htbhf/page/BasePage.java]: I don't understand what the string is doing. Is this selecting multiple classnames? or is this some sort of selector (omitting `.` for each classname)? If it is a selector, could it just be:
```
govuk-radios__label
```

File: src/test/java/uk/gov/dhsc/htbhf/page/BasePage.java [hunk 3]
```diff
@@ -33,16 +37,16 @@ public void open() {
...
-    public WebElement findByClassName(String className) {
-        return webDriver.findElement(By.className(className));
+    public void waitForPageToLoad() {
+        wait.until(ExpectedConditions.titleIs(getPageTitle()));
     }
 
-    public WebElement findByCss(String cssSelector) {
-        return webDriver.findElement(By.cssSelector(cssSelector));
+    public List<WebElement> getRadioButtons() {
+        return webDriver.findElements(By.className("govuk-radios__item"));
```
Comment (gethyn1) [inline src/test/java/uk/gov/dhsc/htbhf/page/BasePage.java]: Do we want these class names to be constants somewhere? Maybe not.

File: src/test/java/uk/gov/dhsc/htbhf/page/EnterDobPage.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    @Override
+    String getPath() {
+        return "/enter-dob";
+    }
+
+    @Override
+    String getPageName() {
+        return "enter date of borth";
```
Comment (gethyn1) [inline src/test/java/uk/gov/dhsc/htbhf/page/EnterDobPage.java]: Typo `borth` -> `birth`

File: src/test/java/uk/gov/dhsc/htbhf/page/BasePage.java (hunk 2 above)
Comment (mikeSmith08) [reply src/test/java/uk/gov/dhsc/htbhf/page/BasePage.java]: Its there by mistake, I'll remove it.

File: src/test/java/uk/gov/dhsc/htbhf/page/BasePage.java (hunk 3 above)
Comment (mikeSmith08) [reply src/test/java/uk/gov/dhsc/htbhf/page/BasePage.java]: I was in two mind about this at they'll probably only be used here, but I'll add them as constants.

File: src/test/java/uk/gov/dhsc/htbhf/page/EnterDobPage.java (hunk 4 above)
Comment (mikeSmith08) [reply src/test/java/uk/gov/dhsc/htbhf/page/EnterDobPage.java]: Yep, will correct, it's not used until I implement the navigation tests so hadn't spotted that yet.

**Pull request merged**